components/root-actions/AccountSettings.js

```diff
@@ -0,0 +1,168 @@
+        {({ id }) => (
+          <CollectivePickerAsync
+            inputId={id}
+            onChange={setSelectedAccountOption}
+            includeDeleted={true}
+            includeArchived={true}
+            value={selectedAccountOption}
+            noCache
+          />
+        )}
+      </StyledInputField>
+
+      <Container px={1} pt={3} pb={3}>
+        <Box pb={2}>
+          <FormattedMessage defaultMessage="Flags" />
+        </Box>
+        <Flex flexWrap="wrap" px={1} mt={2}>
+          <Box pr={4}>
+            <StyledCheckbox
+              name="Deleted"
+              label="Deleted"
+              checked={deletedFlag}
+              onChange={({ checked }) => {
+                setDeletedFlag(checked);
+              }}
+            />
+          </Box>
+          <Box pr={4}>
+            <StyledCheckbox
+              name="Banned"
+              label="Banned"
+              checked={bannedFlag}
+              onChange={({ checked }) => {
+                setBannedFlag(checked);
+              }}
+            />
+          </Box>
+          <Box pr={4}>
+            <StyledCheckbox
+              name="Archived"
+              label="Archived"
+              checked={archivedFlag}
+              onChange={({ checked }) => {
+                setArchivedFlag(checked);
+              }}
+            />
+          </Box>
+          {selectedAccountOption?.value?.isHost && (
+            <Box>
+              <StyledCheckbox
+                name="Trusted Host"
+                label="Trusted Host"
+                checked={trustedHostFlag}
+                onChange={({ checked }) => {
+                  setTrustedHostFlag(checked);
+                }}
+              />
+            </Box>
+          )}
+        </Flex>
+      </Container>
+      <StyledButton
+        mt={4}
+        width="100%"
+        buttonStyle="primary"
+        loading={loading}
+        onClick={async () => {
+          try {
+            const result = await editAccountFlags({
+              variables: {
+                account: { slug: selectedAccountOption?.value?.slug },
+                isBanned: bannedFlag,
+                isDeleted: deletedFlag,
+                isArchived: archivedFlag,
+                isTrustedHost: trustedHostFlag,
+              },
+            });
+            const newSlug = result?.data?.editAccountFlags?.slug;
+            if (newSlug !== selectedAccountOption?.value?.slug) {
+              addToast({
+                type: TOAST_TYPE.SUCCESS,
+                title: 'Success',
+                message: (
+                  <FormattedMessage
+                    defaultMessage="Account has been {deletedFlag, select, true {deleted} other {restored}}. Slug renamed to {newSlug}"
+                    values={{ newSlug, deletedFlag }}
+                  />
+                ),
+              });
+              setSelectedAccountOption([]);
```

Comment:
:ice_cream: For this one, I don't think we need to reset the input after saving. It makes sense to do it for banning because the accounts disappear. Here I would just disable the button when it's already saved.

```suggestion
```